src/main/kotlin/com/pubnub/api/retry/RetryableCallback.kt

```diff
@@ -5,25 +5,23 @@ import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
 import java.util.concurrent.ExecutorService
-import java.util.concurrent.Executors
 import kotlin.time.Duration
 
 internal abstract class RetryableCallback<T>(
     retryConfiguration: RetryConfiguration,
     endpointGroupName: RetryableEndpointGroup,
     private val call: Call<T>,
-    private val isEndpointRetryable: Boolean
+    private val isEndpointRetryable: Boolean,
+    private val executorService: ExecutorService
```

Comment:
you should change this to ScheduledExecutorService and in line 75 call schedule instead of sleep